### PR TITLE
Stopping rules

### LINKED ISSUE #7: Add Inconsistent Policy Simulation
- [x] Allow the user to run a different model at the forward pass (simulation step). Similar to what was done in this [paper](https://ieeexplore.ieee.org/document/9521833). Original issue in https://github.com/andrewrosemberg/HydroPowerModels.jl/issues/54
- [ ] Add stopping criteria that checks: Average Simulation Stalling and Bound Stalling (this can be done by implementing each criteria separately and combining them as described in SDDP.jl). A version was implemented in: 
    - [BoundStalling](https://github.com/andrewrosemberg/SDDP.jl/blob/895a5b91685868c87763c33775b68ae1a37d6d23/src/plugins/stopping_rules.jl#L102)
    - Simulation Stalling
[convergence.jl.zip](https://github.com/LAMPSPUC/HydroPowerModels.jl/files/12566419/convergence.jl.zip)

### PR DESCRIPTION
**Pull request by Thiago-NovaesB**
- Add SimulationStalling
- Add AndStoppingRules (maybe moved into SDDP.jl https://github.com/odow/SDDP.jl/issues/664)
- Add tests

### CODE REVIEW

**Comment by andrewrosemberg** [comment]
It has been a while since I implemented and looked into the stopping criteria we are looking into here. It seems that https://odow.github.io/SDDP.jl/stable/apireference/#SDDP.SimulationStoppingRule is similar to what we want but does rather a simple check instead of a T-test. 

Might be worth to check with @odow if this change is valuable to be added in SDDP.jl.

File: src/plugins/stop_rules.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    p_value = pvalue(htest)
+    sample_mean = Statistics.mean(simulations)
+    
+    if rule.verbose
+        println("Simulated policy value: [", sample_mean, ", ", htest.t, ", ", p_value, "]")
+    end
+
+    rule.previous_simulations = simulations
+    return p_value > rule.p_value
+end
```
Comment (andrewrosemberg) [inline src/plugins/stop_rules.jl]:
```suggestion
end

```

File: Project.toml
Comment (andrewrosemberg) [inline Project.toml:1]: Minor version bump to `0.3.0`.

**Comment by andrewrosemberg** [comment]
Looks good to me once the version is bumped!

I have started running JuliaFormatter after I finish a PR and this could be a good addition here:

```julia
using JuliaFormatter
format("path_to_package_folder", BlueStyle())
```

Since I haven't done it my self in this package (I think), I won't enforce this requirement. In the morning I will merge either way.

File: Project.toml
Comment (Thiago-NovaesB) [reply Project.toml:1]: done

File: src/plugins/stop_rules.jl (hunk 1 above)
Comment (Thiago-NovaesB) [reply src/plugins/stop_rules.jl]: done

**Comment by Thiago-NovaesB** [comment]
Are you going to add it to the package or just githubActions? As it is exclusive to development, it might make sense to only be in actions @andrewrosemberg

**Review by andrewrosemberg — APPROVED** [review]
LGTM

**Comment by andrewrosemberg** [comment]
> Are you going to add it to the package or just githubActions? As it is exclusive to development, it might make sense to only be in actions @andrewrosemberg

I normally run it myself since there are some corner cases where the code breaks after running `format`. What people have is actually a `githubActions` check to see if the PR respects the chosen standard: 
 - https://github.com/NREL-Sienna/PowerSimulations.jl/blob/main/.github/workflows/format-check.yml

**Pull request merged**